common/sagas/config.ts

```diff
@@ -119,25 +119,28 @@ export function* reload(): SagaIterator {
 }
 
 export function* handleNodeChangeIntent(action: ChangeNodeIntentAction): SagaIterator {
-  const currentNode = yield select(getNode);
-  const currentConfig = yield select(getNodeConfig);
-  const currentNetwork = currentConfig.network;
+  const currentNode: string = yield select(getNode);
+  const currentConfig: NodeConfig = yield select(getNodeConfig);
+  const customNets: CustomNetworkConfig[] = yield select(getCustomNetworkConfigs);
+  const currentNetwork =
+    getNetworkConfigFromId(currentConfig.network, customNets) || NETWORKS[currentConfig.network];
+
+  function* bailOut(message: string) {
```

Comment:
Any specific reason for the closure? Seems like this could be re-usable in other sagas.

Reply:
`changeNode(currentNode, currentConfig, currentNetwork)` relies on variables defined in the function. I could move it out, but the behavior of error message + revert node seemed specific to `handleNodeChangeIntent` anyway. Or I could pass them, but then I'd be copying that 3 times.

Reply:
Yeah, I figured all that as well. I don't think there's extreme value by making it re-usable, we can worry about that later.